usecases/linearFunction/test_optimize.py

```diff
@@ -88,23 +106,127 @@ def weighted_all_experiments_model_error(prm_vector):
                       np.dot(me['df'], me['df']) / d['sigma_noise_derivative'] ** 2
             return np.sqrt(obj)
 
-        weighted_start_vector = np.array([0.7])
+        # add all variables 'b' in all model parameter lists as a shared latent variable
+        # return index in the global vector
+        index_b = all_experiments_model_error.latent.add_by_name('b')
+        weighted_start_vector = np.empty([len(all_experiments_model_error.latent)])
+        weighted_start_vector[index_b] = 0.7
+
         weighted_result = least_squares(weighted_all_experiments_model_error, weighted_start_vector)
 
         # assert quadratic component of virtual experiment to be zero
         self.assertAlmostEqual(d['c'], 0.)
         # check to obtain exact linear coefficient
         self.assertAlmostEqual(weighted_result.x[0], d['b'], 4)
 
-    def test_quadratic_virtual_data(self):
+    def test_linear_with_noise_mcmc(self):
+        virtual_experiment_metadata_file = \
+            Path(__file__).parent / 'virtual_experiment_linear_with_noise_mcmc_meta.yaml'
+        with open(virtual_experiment_metadata_file, "r") as f:
+            d = yaml.load(f, Loader=yaml.FullLoader)
+
+        all_experiments_model_error = \
+            MultiLinearModelError('virtual_experiment_linear_with_noise_mcmc_meta.yaml',
+                                  'virtual_experiment_linear_with_noise_mcmc_data.yaml')
+
+        # add noise term as local parameter to all model errors
+        # currently we have to add hyperparameters to the model parameters, since they are not part of the forward model
+        for param_list in all_experiments_model_error.linearModelParameters:
+            param_list.define('sigma_noise_f', 1.)
+            param_list.define('sigma_noise_df', 1.)
```

Comment:
I know, VB does not exist in this test case. But this problem could also be solved with VB, where the noise parameters must not be part of the parameter list. So in the long run, the this "joining" of parameters and hyperparameters for sampling methods must be done somewhere, but IMO later in the process. The initial definition should be independent of the inference method.

That inspired BAMresearch/BayesianInference#11

Reply:
see [here](https://github.com/BAMresearch/BayesianInference/issues/11#issuecomment-776735534)